CMakeLists.txt

```diff
@@ -0,0 +1,190 @@
+cmake_minimum_required(VERSION 3.8.2)
+project(cglm LANGUAGES C)
+
+set(C_STANDARD 11)
+set(C_STANDARD_REQUIRED YES)
+
+set(CGLM_BUILD)
+option(CGLM_SHARED "Shared build" ON)
+option(CGLM_STATIC "Static build" OFF)
+option(CGLM_USE_C99 "" OFF)
+option(CGLM_USE_TEST "Enable Tests" OFF)
+
+if(NOT CGLM_STATIC AND CGLM_SHARED)
+  set(CGLM_BUILD SHARED)
+else(CGLM_STATIC)
+  set(CGLM_BUILD STATIC)
+endif()
+
+if(CGLM_USE_C99)
+  set(C_STANDARD 99)
+endif()
+
+if("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC")
+  list(APPEND LDFLAGS "/W4")
```

Comment:
We have some flags (optimization flag for instance) on Visuals Studio project which you can find in `win/` folder.